src/components/[guild]/JoinModal/JoinModal.tsx

```diff
@@ -52,6 +54,24 @@ const JoinModal = ({ isOpen, onClose }: Props): JSX.Element => {
   if (hasGithubRequirement) allPlatforms.push("GITHUB")
   const allUniquePlatforms = [...new Set(allPlatforms)]
 
+  const allRequirements = roles?.flatMap((role) => role.requirements)
```

Comment:
I'm wondering if we could simplify this logic including the `allUniquePlatforms` part. I've posted a potential simplification here: https://github.com/agoraxyz/guild.xyz/commit/5b3536a7c3e8b3d594e04ae4706b15036e69ae11
Let me know what you think about that approach. I'm not sure if it is better, but I think the current one is hard to read/understand, and somewhat harder to extend

Reply:
Wow, this is a wonderful solution, thanks! Could you please merge it into this branch? And then we could actually merge this to main, if the other parts look good to you!